src/pages/students/edit-student/EditStudent.tsx

```diff
@@ -0,0 +1,135 @@
+
+    const numberOfStepNum = numberOfStep ? parseInt(numberOfStep) : null;
+
+    const { data: studentSaveData } = useGetStudentByIdQuery(studentId!);
+
+    const [updateStudent] = useUpdateStudentMutation();
+
+    const navigate = useNavigate();
+
+    const showStudentData = (studentSaveData: StudentDataProps) => {
+
+        const formattedBirthday = formatDateToISO(studentSaveData.birthday);
+
+        return {
+            name: studentSaveData.name,
+            lastName: studentSaveData.lastName,
+            dni: studentSaveData.dni,
+            cellphoneNumber: studentSaveData.cellphoneNumber,
+            birthday: formattedBirthday,
+            pathologies: studentSaveData.pathologies ?? undefined,
+        };
+    }
+
+    if (!studentSaveData) {
+        return <p>Cargando datos del estudiante...</p>;
+    }
+
+    const handleStudentData = (data: StudentDataProps) => {
+        const updateData = {
+            studentId: studentId!,
+            userId: userId!,
+            ...studentSaveData,
+            ...data,
+            birthday: formatDateToDash(data.birthday),
+
+        }
+        updateStudentData(updateData);
+    }
+
+    const handlePaymentData = (data: PaymentDataProps) => {
+        const updateData = {
+            studentId: studentId!,
+            userId: userId!,
+            ...studentSaveData,
+            ...data,
+            birthday: formatDateToISO(studentSaveData.birthday),
+        }
+        updateStudentData(updateData);
+    }
+
+    /*  const handlePlanData = (data: PlanDataProps) => {
+         const updateData: UpdateStudentRequest = {
+             studentId: studentId!,
+             userId: userId!,
+             ...studentSaveData,
+             ...data,
+            birthday: formatDateToISO(studentSaveData.birthday),
+         }
+         updateStudentData(updateData);
+     } */
+
+    const handleClick = async () => {
+        if (numberOfStepNum === null) return;
+        const refs = [formStudentDataRef, formPaymentDataRef];
+        refs[numberOfStepNum - 1].current.submit();
+    }
+
+
+    const forms: Record<number, { title: string, component: JSX.Element }> = {
+        1: { title: "EDITAR DATOS DEL ALUMNO", component: (<StudentData ref={formStudentDataRef} onSubmit={handleStudentData} data={showStudentData(studentSaveData)} />) },
+        2: { title: "EDITAR DATOS DEL PAGO", component: (<PaymentData ref={formPaymentDataRef} onSubmit={handlePaymentData} data={studentSaveData} actionType="edit" />) },
+        /*  3: <PlanData onSubmit={handlePlanData} data={planData} />, */
+    };
+    if (!numberOfStepNum) {
+        return <p>Paso no válido</p>;
+    }
+
+    const updateStudentData = async (data: UpdateStudentRequest) => {
+        try {
+            await updateStudent(data).unwrap();
+            navigate(-1);
+            toast.success("Los datos del estudiante han sido actualizados");
+        }
+        catch (error) {
+        }
+    }
```

Comment:
Un par de cositas sobre esto:

1. No sirve de nada tener un catch si dentro del bloque no vas a hacer nada al respecto, si haces un trycatch, deberias tener algo que hacer cuando hay un error en la peticion http, como por ejemplo, lanzar un mensaje de error
2. Esto lo podrias modificar de la siguiente manera para hacer uso del unwrap de RTK Query:


```suggestion
    const updateStudentData = (data: UpdateStudentRequest) => {
        updateStudent(data)
            .unwrap()
            .then(() => {
                navigate(-1);
                toast.success("Los datos del estudiante han sido actualizados");
            });
    }
```

3. Me gustó que crearas un método para hacer el llamado y no repetir el código en cada handleData de los forms, bien ahí en darte cuenta =)